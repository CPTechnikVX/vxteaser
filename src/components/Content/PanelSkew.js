import React      from 'react';
import PropTypes  from 'prop-types';
import Constants  from '../../utils/Constants';
import classnames from 'classnames';
import ClickEvent from '../../utils/ClickEvent';

/**
 * Skewed panel
 */
export default class PanelSkew extends React.PureComponent {
	render() {
		const {config, link, modifier, src, windowWidth} = this.props;
		const classList                                  = [];
		const styleObj                                   = {};
		let onClickFn;

		classList.push(Constants.ClassName.PanelSkew);

		if (link) {
			onClickFn = (e) => {
				e.preventDefault();
				e.stopPropagation();

				this.props.onClickFn(new ClickEvent(e, this, {
					link: this.props.link,
				}));
			};
		}

		if (modifier) {
			classList.push(modifier);
		}

		if (windowWidth < config.fixedHeights[1]['greaterThan']) {
			styleObj.top = `-${config.fixedHeights[1]['height']}px`;
		}

		return <a className={classnames(classList)} onClick={onClickFn} style={styleObj}>
			<img className="vxteaser-panel--skew__image" src={src} alt="" />
		</a>;
	}
}

PanelSkew.propTypes = {
	/** @ignore */
	config:      PropTypes.object,
	/** @ignore */
	children:    PropTypes.node,
	/**
	 * Textual color definition
	 */
	color:       PropTypes.oneOf(['primary']),
	link:        PropTypes.string,
	modifier:    PropTypes.string,
	/** @ignore */
	onClickFn:    PropTypes.func,
	src:         PropTypes.string,
	/** @ignore */
	windowWidth: PropTypes.number,
};
