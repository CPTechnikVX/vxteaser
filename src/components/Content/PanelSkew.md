```html
<PanelSkew src="https://www.example.com/img/background.jpg" link="https://www.example.com"></PanelSkew>
```
