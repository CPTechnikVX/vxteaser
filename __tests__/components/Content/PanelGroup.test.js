import React     from 'react';
import {shallow} from 'enzyme';
import PanelSkew from '../../../src/components/Content/PanelSkew';

describe('render panel skew', () => {
	const config = {
		fixedHeights: [
			{},
			{
				greaterThan: 1200,
				height:      260,
			},
		],
	};

	test('fullhd', () => {
		const wrapper = shallow(
			<PanelSkew link="http://www.example.com"
			           modifier="first"
			           src="http://www.example.com/img/default.jpg"
			           windowWidth={1920} config={config}
			>Content</PanelSkew>
		);
		expect(wrapper).toMatchSnapshot();
	});
	test('tablet', () => {
		const wrapper = shallow(
			<PanelSkew link="http://www.example.com"
			           modifier="first"
			           src="http://www.example.com/img/default.jpg"
			           windowWidth={768} config={config}
			>Content</PanelSkew>
		);
		expect(wrapper).toMatchSnapshot();
	});
});
