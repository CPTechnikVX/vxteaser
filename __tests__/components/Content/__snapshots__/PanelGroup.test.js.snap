// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render panel skew fullhd 1`] = `
<a
  className="vxteaser-panel--skew first"
  onClick={[Function]}
  style={Object {}}
>
  <img
    alt=""
    className="vxteaser-panel--skew__image"
    src="http://www.example.com/img/default.jpg"
  />
</a>
`;

exports[`render panel skew tablet 1`] = `
<a
  className="vxteaser-panel--skew first"
  onClick={[Function]}
  style={
    Object {
      "top": "-260px",
    }
  }
>
  <img
    alt=""
    className="vxteaser-panel--skew__image"
    src="http://www.example.com/img/default.jpg"
  />
</a>
`;
